crates/tiny/src/main.rs

```diff
@@ -178,3 +193,41 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth) -> Result<SASLAuth, String> {
+    match sasl_config {
+        config::SASLAuth::Plain { username, password } => Ok(SASLAuth::Plain {
+            username: username.to_string(),
+            password: password.to_string(),
+        }),
+        config::SASLAuth::External { pem } => {
+            // TLS must be on for EXTERNAL
+            if !tls {
+                Err("TLS not enabled".to_string())
+            } else {
+                // load in a cert and private key for TLS client auth
+                match File::open(pem) {
+                    Ok(file) => {
+                        let mut buf = BufReader::new(file);
+                        // extract certificate
+                        let cert = rustls_pemfile::certs(&mut buf)
+                            .map_err(|e| format!("Could not parse pkcs8 PEM: {}", e))?
+                            .pop()
+                            .ok_or("Cert PEM must have one cert")?;
```

Comment:
```suggestion
                            .ok_or("Cert PEM must have at least one cert")?;
```

Since you use `pop` here it's OK to have more than one.